eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use eZ\Publish\SPI\Search\Indexing;
+use eZ\Publish\SPI\Search\Indexing\ContentIndexing;
+use eZ\Publish\SPI\Search\Indexing\LocationIndexing;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use RuntimeException;
+use PDO;
+
+class ReindexCommand extends ContainerAwareCommand
+{
+    const CONTENTOBJECT_TABLE = 'ezcontentobject';
+    const CONTENTOBJECT_TREE_TABLE = 'ezcontentobject_tree';
+
+    /**
+     * @var \eZ\Publish\SPI\Search\Indexing\ContentIndexing | \eZ\Publish\SPI\Search\Indexing\LocationIndexing
+     */
+    private $searchHandler;
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Handler
+     */
+    private $persistenceHandler;
+    /**
+     * @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler
+     */
+    private $databaseHandler;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    private $output;
+
+    /**
+     * Initialize objects required by {@see execute()}.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    public function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->logger = $this->getContainer()->get('logger');
+        $this->searchHandler = $this->getContainer()->get('ezpublish.spi.search');
+        $this->persistenceHandler = $this->getContainer()->get('ezpublish.api.persistence_handler');
+        $this->databaseHandler = $this->getContainer()->get('ezpublish.connection');
+        if (!$this->searchHandler instanceof Indexing) {
+            throw new RuntimeException(
+                'Expected to find Search Engine Handler implementing Indexing but found something else.'
```

Comment:
Ok. Since we now index Locations while indexing content, I moved relevant code [here](https://github.com/alongosz/ezpublish-kernel/blob/0250f1843ca49783de549218f6ac2af186f79a47/eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php#L102-L105).